tests/rptest/services/compatibility/compat_example.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import sys
+import time
+from ducktape.services.background_thread import BackgroundThreadService
+
+
+class CompatExample(BackgroundThreadService):
+    def __init__(self, context):
+        super(CompatExample, self).__init__(context, num_nodes=1)
+        self._cmds = None
+        self._condition = None
+        self._condition_met = False
+        self._timeout = 10
+        self._node = None
+
+    def _worker(self, idx, node):
+        start_time = time.time()
+        if self._node is None:
+            self._node = node
+
+        #Run the example until the condition is met
+        output_iter = self._node.account.ssh_capture(self._cmds[0])
+        while not self._condition_met and time.time() < start_time + self._timeout:
+            line = next(output_iter)
+            self._condition_met = self._condition(line.rstrip())
+
+    #ok - used to determine if the condition is met
+    def ok(self):
+        return self._condition_met
+
+    #set_example - assigns the command, stop condition, and node
+    #for an example
+    def set_example(self, cmds, condition, node=None):
+        self._cmds = cmds
+        self._condition = condition
+        self._condition_met = False
+        self._node = node
+
+    def stop_node(self, node):
+        if self._cmds[1]:
+            node.account.kill_process(self._cmds[1], clean_shutdown=False)
```

Comment:
I'm uncertain why not kill process for `cmds[0]`?

Reply:
cmds[0] is the entire command + options such as `/path/to/cmd --op1 arg1 --op2 arg2`
cmds[1] is just the command string such as `cmd`

I was following a pattern I found in some test examples where `kill_process` is run on the command name instead of the entire command string (which includes the command options). For example https://github.com/vectorizedio/redpanda/blob/f93eed1c6888466993107a558b2abde318151720/tests/rptest/services/kaf_producer.py#L29

I could parse the command string for the relevant sub-string but that seemed unnecessary since we know the command name ahead of time.